src/include/sof/spi.h

```diff
@@ -0,0 +1,197 @@
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *   * Neither the name of the Intel Corporation nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Author: Zhigang Wu <zhigang.wu@intel.com>
+ */
+
+#ifndef __INCLUDE_SPI_H__
+#define __INCLUDE_SPI_H__
+
+enum IPC_STATE {
+	SUE_IPC_IDLE		= 0,
+	SUE_IPC_WRITE_DATA,	/* receive the data from HOST */
+	SUE_IPC_READ_DATA,	/* transmit the data to HOST */
+	SUE_IPC_ERROR,
+};
+
+#define SUE_SPI_READ		1
+#define SUE_SPI_WRITE		0
+
+#define SUE_SPI_BASEADDRESS(x)	(x + 0x00080000)
+#define	SUE_SPI_REG_CTRLR0		SUE_SPI_BASEADDRESS(0x00)
+#define	SUE_SPI_REG_CTRLR1		SUE_SPI_BASEADDRESS(0x04)
+#define	SUE_SPI_REG_SSIENR		SUE_SPI_BASEADDRESS(0x08)
+#define	SUE_SPI_REG_MWCR		SUE_SPI_BASEADDRESS(0x0C)
+#define	SUE_SPI_REG_SER			SUE_SPI_BASEADDRESS(0x10)
+#define	SUE_SPI_REG_BAUDR		SUE_SPI_BASEADDRESS(0x14)
+#define	SUE_SPI_REG_TXFTLR		SUE_SPI_BASEADDRESS(0x18)
+#define	SUE_SPI_REG_RXFTLR		SUE_SPI_BASEADDRESS(0x1C)
+#define	SUE_SPI_REG_TXFLR		SUE_SPI_BASEADDRESS(0x20)
+#define	SUE_SPI_REG_RXFLR		SUE_SPI_BASEADDRESS(0x24)
+#define	SUE_SPI_REG_SR			SUE_SPI_BASEADDRESS(0x28)
+#define	SUE_SPI_REG_IMR			SUE_SPI_BASEADDRESS(0x2C)
+#define	SUE_SPI_REG_ISR			SUE_SPI_BASEADDRESS(0x30)
+#define	SUE_SPI_REG_RISR		SUE_SPI_BASEADDRESS(0x34)
+#define	SUE_SPI_REG_TXOICR		SUE_SPI_BASEADDRESS(0x38)
+#define	SUE_SPI_REG_RXOICR		SUE_SPI_BASEADDRESS(0x3C)
+#define	SUE_SPI_REG_RXUICR		SUE_SPI_BASEADDRESS(0x40)
+#define	SUE_SPI_REG_ICR			SUE_SPI_BASEADDRESS(0x48)
+#define	SUE_SPI_REG_DMACR		SUE_SPI_BASEADDRESS(0x4C)
+#define	SUE_SPI_REG_DMATDLR		SUE_SPI_BASEADDRESS(0x50)
+#define	SUE_SPI_REG_DMARDLR		SUE_SPI_BASEADDRESS(0x54)
+#define	SUE_SPI_REG_DR			SUE_SPI_BASEADDRESS(0x60)
+#define	SUE_SPI_REG_SPICTRLR0	SUE_SPI_BASEADDRESS(0xF4)
+
```

Comment:
This is generic DW IP, so we can remove the SUE_ and use platform data to set the base address.